sklearn/iris_sklearn_trainer/iris.py

```diff
@@ -30,29 +48,41 @@
                        iris_target_filename], stderr=sys.stdout)
 # [END download-data]
 
-
 # [START load-into-pandas]
 # Load data into pandas
 iris_data = pd.read_csv(iris_data_filename).values
 iris_target = pd.read_csv(iris_target_filename).values
 iris_target = iris_target.reshape((iris_target.size,))
+# Split the input data into training and test sets.
+# A fixed random_state will always return the same split result.
+train_data, test_data, train_target, test_target = train_test_split(
+    iris_data, iris_target, test_size=0.4, random_state=0)
 # [END load-into-pandas]
 
-
 # [START train-and-save-model]
 # Train the model
-classifier = svm.SVC(verbose=True)
-classifier.fit(iris_data, iris_target)
-
+classifier = svm.SVC(verbose=True, kernel=args.kernel, C=args.c)
+classifier.fit(train_data, train_target)
 # Export the classifier to a file
 model = 'model.joblib'
 joblib.dump(classifier, model)
 # [END train-and-save-model]
 
+# [START report-hyperparameter-tuning-metrics]
+# Calculate the mean accuracy on the given test data and labels.
+score = classifier.score(test_data, test_target)
+# Report the mean accuracy as hyperparameter tuning objective metric.
+import hypertune
+hpt = hypertune.HyperTune()
+hpt.report_hyperparameter_tuning_metric(
+    hyperparameter_metric_tag='my_metric_tag',
+    metric_value=score,
+    global_step=1)
+# [END report-hyperparameter-tuning-metrics]
 
 # [START upload-model]
 # Upload the saved model file to Cloud Storage
-model_path = os.path.join('gs://', bucket, datetime.datetime.now().strftime(
-    'iris_%Y%m%d_%H%M%S'), model)
+model_path = os.path.join(args.job_dir, model)
 subprocess.check_call(['gsutil', 'cp', model, model_path], stderr=sys.stdout)
```

Comment:
Please update to use the python google cloud storage package: `from google.cloud import storage`

Similar to this flow from: https://github.com/GoogleCloudPlatform/cloudml-samples/blob/master/sklearn/notebooks/ml_engine_training.ipynb
```
# Upload the model to GCS
bucket = storage.Client().bucket(BUCKET_ID)
blob = bucket.blob('{}/{}'.format(
    datetime.datetime.now().strftime('census_%Y%m%d_%H%M%S'),
    model))
blob.upload_from_filename(model)
```